pkg/awsutils/awsutils.go

```diff
@@ -674,17 +689,50 @@ func (cache *EC2InstanceMetadataCache) AllocIPAddress(eniID string) error {
 	return nil
 }
 
+// GetENIipLimit return IP address limit per ENI based on EC2 instance type
+func (cache *EC2InstanceMetadataCache) GetENIipLimit() (int64, error) {
```

Comment:
I note there's a lot of inconsistency in this code on the capitalization of acronyms and initialisms.  `GetENIipLimit` vs `InstanceIPsAvailable` vs `AssignPrivateIpAddresses`, vs `addENIaddressesToDataStore` and more.  The golang style recommendation from various sites and by example in stdlib is to [allcaps acronyms and initialism](https://github.com/golang/go/wiki/CodeReviewComments#initialisms) except of course the special case of private vs public leading character requirements.